radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/handler/DeveloperHandler.java

```diff
@@ -128,14 +131,28 @@ public JSONObject handleParseTxn(JSONObject request) {
 	public JSONObject handleParseAddress(JSONObject request) {
 		return withRequiredParameters(
 			request,
-			List.of("address"),
+			List.of("address", "type"),
 			params -> Result.wrap(
 				e -> Failure.failure(-1, e.getMessage()),
 				() -> {
+					var type = params.getString("type");
 					var address = params.getString("address");
-					var p = AccountAddressing.parseUnknownHrp(address);
-					var hrp = p.getFirst();
-					var pubKey = p.getSecond().publicKey().orElseThrow();
+					Pair<String, ECPublicKey> pair;
+					switch (type) {
+						case "account":
+							pair = AccountAddressing.parseUnknownHrp(address).mapSecond(addr -> addr.publicKey().orElseThrow());
+							break;
+						case "node":
+							pair = NodeAddressing.parseUnknownHrp(address);
+							break;
+						case "validator":
+							pair = ValidatorAddressing.parseUnknownHrp(address);
+							break;
+						default:
+							throw new IllegalArgumentException("type must be: [account|node|validator]");
+					}
```

Comment:
I think code will be much more readable if `switch` will be extracted into dedicated method.